service/src/test/java/com/codeforcommunity/processor/ProtectedUserProcessorImplTest.java

```diff
@@ -0,0 +1,178 @@
+import org.jooq.generated.Tables;
+import org.jooq.generated.tables.records.TeamRecord;
+import org.jooq.generated.tables.records.UserTeamRecord;
+import org.jooq.generated.tables.records.UsersRecord;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ProtectedUserProcessorImplTest {
+  // the JooqMock to use for testing
+  JooqMock mockDb;
+  // the ProcessorImpl to use for testing
+  ProtectedUserProcessorImpl processor;
+
+  /** Method to setup mockDb and processor. */
+  @BeforeEach
+  void setup() {
+    this.mockDb = new JooqMock();
+    this.processor = new ProtectedUserProcessorImpl(mockDb.getContext());
+  }
+
+  // successfully deletes user when user is a member
+  @Test
+  public void testDeleteUser1() {
+    UserTeamRecord myUserTeam = mockDb.getContext().newRecord(Tables.USER_TEAM);
+    myUserTeam.setUserId(1);
+    myUserTeam.setTeamId(5);
+    myUserTeam.setTeamRole(TeamRole.MEMBER);
+    mockDb.addReturn("SELECT", myUserTeam);
+    mockDb.addEmptyReturn("DELETE");
+
+    JWTData jwtData = new JWTData(1, PrivilegeLevel.STANDARD);
+
+    this.processor.deleteUser(jwtData);
+    assertEquals(1, mockDb.timesCalled("SELECT"));
+    assertEquals(3, mockDb.timesCalled("DELETE"));
+    // user_id
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(0)[0]);
+    // user_id from user_team
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(1)[0]);
+    // user_id from users
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(2)[0]);
+    // user_id from user_team
+    assertEquals(1, mockDb.getSqlBindings().get("SELECT").get(0)[0]);
+  }
+
+  // successfully deletes user when user is a leader
+  @Test
+  public void testDeleteUser2() {
+    UserTeamRecord myUserTeam = mockDb.getContext().newRecord(Tables.USER_TEAM);
+    myUserTeam.setUserId(1);
+    myUserTeam.setTeamId(5);
+    myUserTeam.setTeamRole(TeamRole.LEADER);
+    mockDb.addReturn("SELECT", myUserTeam);
+    mockDb.addEmptyReturn("DELETE");
+
+    TeamRecord myTeamRecord = mockDb.getContext().newRecord(Tables.TEAM);
+    myTeamRecord.setId(5);
+    myTeamRecord.setName("Team 5");
+    myTeamRecord.setBio("bio");
+    myTeamRecord.setGoal(2);
+    Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+    myTeamRecord.setGoalCompletionDate(timestamp);
+    mockDb.addReturn("SELECT", myTeamRecord);
+    mockDb.addEmptyReturn("DELETE");
+
+    JWTData jwtData = new JWTData(1, PrivilegeLevel.STANDARD);
+
+    this.processor.deleteUser(jwtData);
+    assertEquals(1, mockDb.timesCalled("SELECT"));
+    assertEquals(4, mockDb.timesCalled("DELETE"));
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(0)[0]);
+    assertEquals(5, mockDb.getSqlBindings().get("DELETE").get(1)[0]);
+    assertEquals(1, mockDb.getSqlBindings().get("DELETE").get(2)[0]);
+    assertEquals(5, mockDb.getSqlBindings().get("DELETE").get(3)[0]);
+    assertEquals(1, mockDb.getSqlBindings().get("SELECT").get(0)[0]);
```

Comment:
Can you add descriptions like you did above (here and everywhere else there are a bunch of grouped `getSqlBindings` like this)?

Reply:
Got it